torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1208,13 +1217,26 @@ c10::intrusive_ptr<ProcessGroup::Work> ProcessGroupNCCL::pointToPoint(
     work->store_ = store_;
   }
 
-  if (opType == OpType::RECV) {
+
+  // Future only needs to be created and marked completed with outputs for
+  // recv(), but still create future for use cases such as profiling even for
+  // send().
+  {
     at::cuda::CUDAMultiStreamGuard streamGuard(ncclStreams_[key]);
     work->future_ = c10::make_intrusive<at::cuda::CUDAFuture>(
         c10::ListType::create(c10::TensorType::get()));
     work->future_->markCompleted(at::IValue(*work->outputs_));
```

Comment:
Will this change the behavior? If this is `OpType::SEND`, what's in `*work->outputs_`?

Reply:
For `send()` it will be the input tensors. I think previously, we didn't have this future at all which is sort of strange because ideally for nccl, `work.get_future()` should return a future regardless of the type of work, but before this it would be an "empty" future. 

With this PR, if we do:
```
tensor = torch.tensor(...)
fut = ncclSend(tensor).get_future()
fut.wait() # will just return tensor unmodified
```

Actually, we should probably add some tests for `get_future()` for p2p nccl collectives cc @agolynski

Reply:
Is this behavior documented anywhere?

Reply:
We don't document `get_future` anywhere and only mention it for gradient compression currently: https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html?highlight=get_future. I think after future/work merge and `get_future()` API having support for all backends, we should add documentation for this. cc @agolynski

Reply:
Filed https://github.com/pytorch/pytorch/issues/57646 for this